apis/v1alpha1/tcproute_types.go

```diff
@@ -36,9 +36,12 @@ type TCPRoute struct {
 // TCPRouteSpec defines the desired state of TCPRoute
 type TCPRouteSpec struct {
 	// Rules are a list of TCP matchers and actions.
+	//
+	// +kubebuilder:validation:MaxItems=16
```

Comment:
Should we have `MinItems=1` here as well? What good is a TCPRoute without any rules? I understand matches being optional, but aren't rules the only way to forward traffic anywhere?